background.js

```diff
@@ -0,0 +1,24 @@
+function onBeforeRequest(data){
+    //perhaps put 'data' in parameter
+    console.log("top of on before request method");
+    console.log(data);
+    //need to return something
+    //this is where blocking goes and also figuring out whether something needs to be blocked
+    var urlNotFixed = window.location.toString();
+    urlNotFixed.replace(" ", "");
+    urlNotFixed.toLowerCase();
+    var urlFixed = urlNotFixed;
+    console.log(urlFixed);
+    var blockedList = "facebook.com";
+    if(urlFixed.contains(blockedList)){
+        return webRequest.BlockingResponse;
+    }
+
+    //for block list: look at string mdn on google and look at the methods for the string object
+    
+}
+console.log("executing background code");
+chrome.webRequest.onBeforeRequest.addListener(onBeforeRequest);
```

Comment:
As you can glean from the error message that'll get dumped to the console when this code executes, a required argument is missing.....the `onBeforeRequest` docs will tell you which one it is (any argument not marked 'optional' can be assumed to be required). Then, that required argument itself has one property that is required, so you'll need to add that too.